api/access/server.go

```diff
@@ -160,12 +160,97 @@ func (s *Server) SendTransaction(ctx context.Context, in *access.SendTransaction
 	return nil, errors.New("not implemented")
 }
 
-func (s *Server) GetTransaction(ctx context.Context, in *access.GetTransactionRequest) (*access.TransactionResponse, error) {
-	return nil, errors.New("not implemented")
+func (s *Server) GetTransaction(_ context.Context, in *access.GetTransactionRequest) (*access.TransactionResponse, error) {
+	id := flow.HashToID(in.Id)
+	tb, err := s.index.Transaction(id)
+	if err != nil {
+		return nil, fmt.Errorf("could not retrieve transaction: %w", err)
+	}
+
+	resp := access.TransactionResponse{
+		Transaction: convert.TransactionToMessage(*tb),
+	}
+
+	return &resp, nil
 }
 
-func (s *Server) GetTransactionResult(ctx context.Context, in *access.GetTransactionRequest) (*access.TransactionResultResponse, error) {
-	return nil, errors.New("not implemented")
+func (s *Server) GetTransactionResult(_ context.Context, in *access.GetTransactionRequest) (*access.TransactionResultResponse, error) {
+	txID := flow.HashToID(in.Id)
+	tx, err := s.index.Transaction(txID)
+	if err != nil {
+		return nil, fmt.Errorf("could not retrieve transaction: %w", err)
+	}
+
+	result, err := s.index.Result(txID)
+	if err != nil {
+		return nil, fmt.Errorf("could not retrieve transaction result: %w", err)
+	}
+
+	// The code below is a simplified and limited implementation to derive the transaction status from
+	// https://github.com/onflow/flow-go/blob/v0.17.4/engine/access/rpc/backend/backend_transactions.go#L257-L328
+	statusCode := uint32(0)
+	if result.ErrorMessage == "" {
+		statusCode = 1
+	}
+
+	// First, we need to retrieve the last indexed height to access the latest seals.
+	lastHeight, err := s.index.Last()
+	if err != nil {
+		return nil, fmt.Errorf("could not retrieve last height: %w", err)
+	}
+
+	sealIDs, err := s.index.SealsByHeight(lastHeight)
+	if err != nil {
+		return nil, fmt.Errorf("could not retrieve block seals: %w", err)
+	}
+
+	// We then go through all sealed blocks at the last height and look for the latest sealed height.
+	var sealedHeight uint64
+	for _, sealID := range sealIDs {
+		seal, err := s.index.Seal(sealID)
+		if err != nil {
+			return nil, fmt.Errorf("could not retrieve seal: %w", err)
+		}
+
+		h, err := s.index.HeightForBlock(seal.BlockID)
+		if err != nil {
+			return nil, fmt.Errorf("could not retrieve block height: %w", err)
+		}
+
+		if h > sealedHeight {
+			sealedHeight = h
+		}
+	}
+
+	// We also need the height of the transaction we're looking at.
+	height, err := s.index.HeightForBlock(tx.ReferenceBlockID)
```

Comment:
This is incorrect. As I wrote on Slack, the `ReferenceBlockID` has nothing to do with the block that includes the transaction. It's merely used for expiry of the transaction. As far as I know, there is no way to get the block for a transaction in such a way, so we will have to make one.